src/Integration.Vsix.UnitTests/SonarLintDaemon/SonarLintDaemonTests.cs

```diff
@@ -58,24 +58,86 @@ public void SetUp()
         [TestCleanup]
         public void CleanUp()
         {
+            testableDaemon.Dispose();
             ForceDeleteDirectory(tempPath);
             ForceDeleteDirectory(storagePath);
         }
 
+        [TestMethod]
+        public void IsRunning_NotRunningOnCreation()
+        {
+            testableDaemon.IsRunning.Should().BeFalse();
+        }
 
         [TestMethod]
-        public void Not_Installed()
+        public void IsInstalled()
         {
+            // Sanity check - not expecting the exe directory to exist to start with
+            var exeDirectory = Path.GetDirectoryName(testableDaemon.ExePath);
+            Directory.Exists(exeDirectory).Should().BeFalse();
+
+            // 1. No directory or file -> false
             testableDaemon.IsInstalled.Should().BeFalse();
-            testableDaemon.IsRunning.Should().BeFalse();
+
+            // 2. Directory and file exist -> true
+            Directory.CreateDirectory(exeDirectory);
+            File.WriteAllText(testableDaemon.ExePath, "junk");
+            testableDaemon.IsInstalled.Should().BeTrue();
         }
 
         [TestMethod]
         public void Run_Without_Install()
         {
             Action op = () => testableDaemon.Start();
 
-            op.Should().ThrowExactly<InvalidOperationException>();
+            op.Should().ThrowExactly<InvalidOperationException>().And.Message.Should().Be("Daemon is not installed");
+        }
+
+        [TestMethod]
+        public void Run_With_Install_Succeeds()
+        {
+            testableDaemon.Port.Should().Be(0);
+
+            // Fake the installation i.e. create the directory and file to execute.
+            testableDaemon.SetUpDummyInstallation("test1.bat",
+@"@echo Hello world
+xxx yyy
```

Comment:
We can't launch the real daemon from the tests, so I've added some override points to allow the tests to substitute an alternative batch file to run instead. That allows us to check that the process is launched, and also that the standard and error outputs are captured correctly.